src/main/java/org/spongepowered/api/util/Color.java

```diff
@@ -0,0 +1,204 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.util;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import org.apache.commons.lang3.Validate;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataSerializable;
+import org.spongepowered.api.data.MemoryDataContainer;
+import org.spongepowered.api.data.Queries;
+import org.spongepowered.api.data.type.DyeColor;
+
+import java.util.Objects;
+
+public final class Color implements DataSerializable {
+
+    private static final int MASK = 0xff;
+
+    public static final Color BLACK = ofHex(0x000000);
+
+    public static final Color GRAY = ofHex(0x808080);
+
+    public static final Color WHITE = ofHex(0xFFFFFF);
+
+    public static final Color BLUE = ofHex(0x0000FF);
+
+    public static final Color GREEN = ofHex(0x008000);
+
+    public static final Color LIME = ofHex(0x00FF00);
+
+    public static final Color RED = ofHex(0xFF0000);
+
+    public static final Color YELLOW = ofHex(0xFFFF00);
+
+    public static final Color MAGENTA = ofHex(0xFF00FF);
+
+    public static final Color PURPLE = ofHex(0xAA00FF);
+
+    public static final Color DARK_CYAN = ofHex(0x008B8B);
+
+    public static final Color DARK_GREEN = ofHex(0x006400);
+
+    public static final Color DARK_MAGENTA = ofHex(0x8B008B);
+
+    public static final Color NAVY = ofHex(0x000080);
+
+    public static final Color UNKNOWN = ofHex(0xFF00AA);
+
+    public static Color ofHex(int hex) {
+        return ofRGB(hex >> 16 & MASK, hex >> 8 & MASK, hex & MASK);
+    }
+
+    public static Color ofRGB(int red, int green, int blue) {
+        return new Color(red, green, blue);
+    }
+
+    public static Color ofJavaColor(java.awt.Color color) {
+        return new Color(color.getRed(), color.getGreen(), color.getBlue());
+    }
+
+    public static Color mixDyeColors(DyeColor... colors) {
+        Validate.noNullElements(colors, "No nulls allowed!");
+        final Color[] actualColors = new Color[colors.length];
+        for (int i = 0; i < colors.length; i++) {
+            actualColors[i] = colors[i].getColor();
+        }
+
+        return mixColors(actualColors);
+    }
+
+    public static Color mixColors(Color... colors) {
+        Validate.noNullElements(colors, "No null colors allowed!");
+        checkArgument(colors.length > 0, "Cannot have an empty color array!");
+        if (colors.length == 1) {
+            return colors[0];
+        } else {
+            int red = colors[0].getRed();
+            int green = colors[0].getGreen();
+            int blue = colors[0].getBlue();
+            for (int i = 1; i < colors.length; i++) {
+                red += colors[i].getRed();
+                green += colors[i].getGreen();
+                blue += colors[i].getBlue();
+            }
+            double averageRed = red / colors.length;
```

Comment:
No alpha mixing? This should be (in functional pseudo code) `colors.sum(color -> color.rgb * color.a) / colors.sum(color -> color.a)`